src/bfloat16.jl

```diff
@@ -264,8 +264,9 @@ function ==(x::BFloat16, y::BFloat16)
     return ix == iy
 end
 
-for op in (:<, :<=, :>, :>=, :!=)
-    @eval ($op)(a::BFloat16, b::BFloat16) = ($op)(Float32(a), Float32(b))
+<(a::BFloat16, b::BFloat16) = Float32(a) < Float32(b)
+@static if VERSION < v"1.8"
```

Comment:
why don't we need this on newer versions?

Reply:
The real question is why is this needed on 1.6/1.7 and after a bit of digging, it's because of https://github.com/JuliaLang/julia/issues/46327, fixed for 1.8 by https://github.com/JuliaLang/julia/pull/46341.

As a workaround, we need to define our own `<=` for BFloat16 in Julia versions below 1.8 otherwise `(<=)(x::T, y::T) where {T<:Real}` in promotion.jl is more specific than the (correct) fallback in operators.jl